### DISCUSSION TITLE
x-on not working in table and text/div elements. Tested on Chrome 86.0.4240

### DISCUSSION THREAD

**Discussion by gaslit** [alpinejs/alpine#876] [1. Help]
Yes im aware that using tables is... passe. But please bear with me.

Is there any way I can fix using alpinejs in table elements?

```
<table x-on:click="console.log('Clicked');" style="width: 100%;" width="100%" cellpadding="0" cellspacing="0">
<tr x-on:click="console.log('Clicked');">
  <td style="text-align: left; width: 80px;" width="80" align="left">
    <img  x-on:click="console.log('Clicked');" src="" width="80" alt="Image" style="border: 0; max-width: 100%; line-height: 100%; vertical-align: middle;">
  </td>
</tr>
</table>
```

All does not work.

On a seperate note,

`<h1 x-on:click="console.log('Clicked');"  class="italic font-bold text-center text-orange-500 text-10xl font-display">Testing</h1>`

The above does not work too and changing it to a div also proved fruitless.

Many thanks.

**Comment by ryangjchandler** [comment]
@gaslit Are these elements inside of an element with and `x-data` attribute?

**Reply by gaslit** [reply]
No. Just within `<body>` tags.

And there is no x-data attrib for all the tags including the <body> itself

**Reply by HugoDF** [reply]
@gaslit x-on won't work outside of an alpine component (ie. If it's not a child of an element with x-data on it)

**Reply by gaslit** [reply]
> @gaslit x-on won't work outside of an alpine component (ie. If it's not a child of an element with x-data on it)

Thanks so much HugoDF. I misunderstood how alpinejs worked. I have enclosed elements within a parent elem with x-data defined and it now works.

If you have any recommendations on books or materials on alpinejs, please let me know. It is indeed a magnificent tool. :)

**Reply by HugoDF** [reply]
@gaslit I run a weekly newsletter with Alpine resources https://alpinejs.codewithhugo.com/newsletter/
